.github/workflows/pytest-core-nompi.yml

```diff
@@ -124,18 +135,18 @@ jobs:
     - name: Build docker image
       if: contains(matrix.name, 'docker')
       run: |
-          docker build . --file docker/Dockerfile.devito --tag devito_img --build-arg base=devitocodes/bases:cpu-${{ matrix.arch }}
+          docker build . --file docker/Dockerfile.devito --tag devito_img --build-arg base=devitocodes/bases:cpu-${{ matrix.tag }}
 
     - name: Set run prefix
       run: |
           if [[ "${{ matrix.name }}" =~ "docker" ]]; then
-              echo "RUN_CMD=docker run --rm -e CODECOV_TOKEN=${{ secrets.CODECOV_TOKEN }} --name testrun devito_img"  >> $GITHUB_ENV
+              echo "RUN_CMD=docker run --rm -e CODECOV_TOKEN=${{ secrets.CODECOV_TOKEN }} -e DEVITO_ARCH=${{ matrix.arch }} --name testrun devito_img"  >> $GITHUB_ENV
```

Comment:
no, this is (has to be) encoded in the Dockerfile already

Reply:
Well no the dockerfile and image are exactly the same for `icc` and `icx` with the exception of this single env var so it's quite a pain to have to rebuild a whole image in the build CI just to change this one var that can be set at runtime easily like that.

Reply:
I think I can consider this resolved, as we only want to build on image and use this env var to distinguish between ICC and icx